components/tabs/src/storage.rs

```diff
@@ -362,6 +378,32 @@ impl TabsStorage {
     }
 }
 
+// Trim the amount of tabs in a list to fit the specified memory size
+fn trim_tabs_length(
+    tabs: &mut Vec<RemoteTab>,
+    payload_size_max_bytes: usize,
+) -> Result<Vec<RemoteTab>> {
```

Comment:
Do we need to return the Vec if we are modifying it in places by passing a mutable reference? We might be able to get away by either:

- pass in an owned mut Vec, and return the same one back
- keep the mutable reference, but don't clone at the end and have the caller consume what was borrowed

Reply:
yeah, thanks - I agree, let's do this (ie, drop the result vec and just re-use the (not trimmed) arg we passed in

Reply:
Great suggestion @tarikeshaq, done!